### DISCUSSION TITLE
I am trying to add Prometheus client to my prompt flow, is there a recommended way how to do that?

### DISCUSSION THREAD

**Discussion by niveditau** [microsoft/promptflow#2021] [Q&A] [answered]
My goal is to expose the /metrics endpoint.

**Comment by brynn-code** [comment]
Hi, sorry for the late response, we are not support /metrics for now, would you like to share more details so that we could better evaluate the requirements? For example, what metrics would you like to collect? I think we are talking about the serving scenario, not the flow authoring, is it correct?

**Reply by niveditau** [reply, marked as answer]
No worries at all, I figured out a work around. To add Prometheus to a Flask application you need to include the prometheus-flask-exporter python package to your flow's dependencies. Then pass the instance of your flask application to PrometheusMetrics class constructor which you will import from prometheus-flask-exporter. This will expose the /metrics endpoint.

**Reply by wxpjimmy** [reply]
@niveditau just to clarify, this flask way only works for single process, if you are serving with gunicorn, you need to follow this guide to enable multiprocess mode: https://prometheus.github.io/client_python/multiprocess/.

If you are using pf serving, could you share with us your scenario and what metrics you want to expose in your flow? We can better evaluate how we can help on this.
